api/client/service/update.go

```diff
@@ -369,23 +372,54 @@ func updateMounts(flags *pflag.FlagSet, mounts *[]swarm.Mount) {
 	*mounts = newMounts
 }
 
-func updatePorts(flags *pflag.FlagSet, portConfig *[]swarm.PortConfig) {
+type byPortConfig []swarm.PortConfig
+
+func (r byPortConfig) Len() int      { return len(r) }
+func (r byPortConfig) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
+func (r byPortConfig) Less(i, j int) bool {
+	// We convert PortConfig into `port/protocol`, e.g., `80/tcp`
+	// In updatePorts we already filter out with map so there is duplicate entries
+	return strings.Compare(portConfigToString(&r[i]), portConfigToString(&r[j])) < 0
```

Comment:
Thanks @stevvooe. Must have short circuited to forget the string comparison. The pull request has been updated.